### PR TITLE
Adding cursor loading from the Resource Bundle on OSX

### PR DESCRIPTION
**Pull request by oneeyeman1**
Hi,

This is an update of https://github.com/wxWidgets/wxWidgets/pull/22391.

I updated the code to the latest master put the latest code from the PR in question

Hopefully the build will succeed this time...

Next will be to create  a unit test for this

### CODE REVIEW

**Comment by oneeyeman1** [comment]
@csomor ,
Can SVG be used as a cursor?

**Review by vadz — COMMENTED** [review]
It looks like this should work, but I didn't test it.

**Comment by oneeyeman1** [comment]
It looks like tests uses samples\sample.rc.

Should there be tests\test.rc or the horse.cur needs to be copied to the sample directory?

Please advise.

File: samples/sample.rc [hunk 1]
```diff
@@ -17,6 +17,8 @@
 // starting with "a"
 aaaaaaaa ICON "sample.ico"
 
+horse    CURSOR "horse.cur"
+
```
Comment (vadz) [inline samples/sample.rc:21]: We really don't want to add it here. Please add it to the test .rc file.

File: tests/Makefile.in [hunk 2]
```diff
@@ -514,6 +514,8 @@ test$(EXEEXT): $(TEST_OBJECTS)
 @COND_PLATFORM_MACOSX_1_USE_GUI_1@	
 @COND_PLATFORM_MACOSX_1_USE_GUI_1@	
 @COND_PLATFORM_MACOSX_1_USE_GUI_1@	cp -f $(top_srcdir)/src/osx/carbon/wxmac.icns test_gui.app/Contents/Resources/wxmac.icns
+@COND_PLATFORM_MACOSX_1_USE_GUI_!@	cp -f horse.cur test_gui.app/Contents/Resources/horse.cur
+@COND_PLATFORM_MACOSX_!_USE_GUI_!@	cp -f horse.png test_gui.app/Contents/Resources/horse.png
```
Comment (vadz) [inline tests/Makefile.in:518]: How did you manage to mistype this so badly?? `!` should be `1`s here, obviously.

File: tests/controls/windowtest.cpp [hunk 3]
```diff
@@ -148,6 +149,16 @@ TEST_CASE_METHOD(WindowTestCase, "Window::Mouse", "[window]")
 
     CHECK(m_window->GetCursor().IsOk());
 
+    wxCursor resCursor1( "horse" );
+    CHECK( resCursor1.IsOk() );
+#ifdef __WXOSX__
+    wxRemoveFile( wxStandardPaths::Get().GetResourcesDir() + "horse.png" );
```
Comment (vadz) [inline tests/controls/windowtest.cpp:155]: Why do you call `wxRemoveFile()` here??

**Review by vadz — CHANGES_REQUESTED** [review]
Sorry but the changes here make no sense at all to me.

File: tests/controls/windowtest.cpp (hunk 3 above)
Comment (oneeyeman1) [reply tests/controls/windowtest.cpp:155]: @vadz ,
The code checks for presence of png and cur file.
I need to check if both cases are covered as well as "no file present"

So I need to remove png and then cur file..

Or you mean it is easier to use simple "rm" here?

File: samples/sample.rc (hunk 1 above)
Comment (oneeyeman1) [reply samples/sample.rc:21]: @vadz 
As mentioned before, there is no such file:

```
Igor@fordanwork MINGW64 /c/wxFork (GetCursorFromBundleResoure_part2)
$ find . -name *\.rc
./demos/bombs/bombs.rc
./demos/forty/forty.rc
./demos/fractal/fractal.rc
./demos/life/life.rc
./demos/poem/wxpoem.rc
./include/wx/msw/wx.rc
./samples/dnd/dnd.rc
./samples/docview/docview.rc
./samples/dragimag/dragimag.rc
./samples/image/image.rc
./samples/listctrl/listtest.rc
./samples/mdi/mdi.rc
./samples/mfc/mfctest.rc
./samples/nativdlg/dialog1.rc
./samples/nativdlg/nativdlg.rc
./samples/ownerdrw/ownerdrw.rc
./samples/regtest/regtest.rc
./samples/sample.rc
./samples/sashtest/sashtest.rc
./samples/sound/sound.rc
./samples/svg/svgtest.rc
./samples/toolbar/toolbar.rc
./src/expat/expat/win32/version.rc
./src/msw/version.rc
./src/png/contrib/visupng/VisualPng.rc
./src/png/scripts/pngwin.rc
./src/zlib/contrib/vstudio/vc10/zlib.rc
./src/zlib/contrib/vstudio/vc11/zlib.rc
./src/zlib/contrib/vstudio/vc12/zlib.rc
./src/zlib/contrib/vstudio/vc14/zlib.rc
./src/zlib/contrib/vstudio/vc9/zlib.rc
./src/zlib/win32/zlib1.rc
./utils/helpview/src/client.rc
./utils/helpview/src/helpview.rc
./utils/hhp2cached/hhp2cached.rc
./utils/screenshotgen/src/screenshotgen.rc
./utils/wxrc/wxrc.rc
```

I can either create one or copy the horse.cur file over to sample directory.

Thank you.

Comment (vadz) [reply samples/sample.rc:21]: Yes, you obviously need to add a `test.rc` file if you want to use the resources in the test. I'm sorry but how exactly do you think it's going to work otherwise?

File: src/osx/carbon/cursor.cpp [hunk 4]
```diff
@@ -278,7 +278,26 @@ wxCursor::wxCursor(const wxString& cursor_file, wxBitmapType flags, int hotSpotX
     if ( flags == wxBITMAP_TYPE_MACCURSOR_RESOURCE )
     {
 #if wxOSX_USE_COCOA
-        wxFAIL_MSG( wxT("Not implemented") );
+        wxImage image;
+        wxFileName fileName( wxStandardPaths::Get().GetResourcesDir() + "/" + cursor_file + ".png" );
```
Comment (vadz) [inline src/osx/carbon/cursor.cpp:282]: This is really hopeless :-(

```suggestion
        wxFileName fileName( wxStandardPaths::Get().GetResourcesDir(), cursor_file, ".png" );
```

File: tests/controls/windowtest.cpp [hunk 5]
```diff
@@ -148,6 +195,21 @@ TEST_CASE_METHOD(WindowTestCase, "Window::Mouse", "[window]")
 
     CHECK(m_window->GetCursor().IsOk());
 
+#ifndef __WXGTK__
+    wxCursor resCursor1( "horse" );
+    CHECK( resCursor1.IsOk() );
+#else
+    wxCursor rresCursor1( (const char *) horse, 32, 32, 16, 0, nullptr, wxWHITE, wxBLACK );
+    CHECK( rresCursor1.IsOk() );
+#endif
```
Comment (vadz) [inline tests/controls/windowtest.cpp:204]: Could we get at least some comments about what's going on here? Why is wxGTK handled specially? Surely creating cursors from the embedded data should work in all ports?

File: tests/controls/windowtest.cpp (hunk 3 above)
Comment (vadz) [reply tests/controls/windowtest.cpp:206]: I mean that you really must explain about what you're doing. You have some very original, to put it mildly, ideas about how tests should work and the least you can do is to carefully document your thinking because I assure you it's not obvious.

In this particular case, it seems -- to anybody else except you, apparently -- obvious that the test shouldn't modify the file system in any permanent way so it should never delete any files not created by it.

It also seems very unclear what exactly is the test testing for because it still expects to create the cursor successfully after removing the file (probably because `wxRemoveFile()` fails because you don't pass it the correct path...).

In short, I have absolutely no idea about what are you trying to do here but I'm pretty sure that what you currently do is wrong.

Please explain in the comments here what is the code supposed to do and/or better think more about this and change it.

File: tests/controls/windowtest.cpp (hunk 5 above)
Comment (oneeyeman1) [reply tests/controls/windowtest.cpp:204]: @vadz ,
Do you want me to quote the documentation here?

**Comment by oneeyeman1** [comment]
I don't understand MSW cross-compiler failure.

I did change makefile.gcc to include the new resource file....

And MSVC build succeeded with the same modification...

File: tests/controls/windowtest.cpp (hunk 5 above)
Comment (oneeyeman1) [reply tests/controls/windowtest.cpp]: @vadz ,
I added some comments explaining the difference between X/GTK and Windows/OSX.

Let me know if this is not enough.

The code in the test can be used as is in the documentation instead of the current one.

File: src/osx/carbon/cursor.cpp (hunk 4 above)
Comment (oneeyeman1) [reply src/osx/carbon/cursor.cpp]: Fixed

File: tests/controls/windowtest.cpp (hunk 3 above)
Comment (oneeyeman1) [reply tests/controls/windowtest.cpp:216]: @vadz ,
I added the comment explaining the reason.

Let me know if this is not enough.

**Comment by vadz** [comment]
Sorry, not going to look at this again until you can at least fix the tests.

**Comment by MaartenBent** [comment]
If I may give a suggestion; don't try to manually modify the makefiles, but update `tests/test.bkl` (and the equivalent CMake files) and then regenerate them. For example, add the resource file similar to how it is done for the samples in https://github.com/wxWidgets/wxWidgets/blob/8b5f0493151f1b74180021256e3ee4b6b2617e9e/build/bakefiles/common_samples_intree.bkl#L37

**Comment by oneeyeman1** [comment]
@vadz , @MaartenBent ,

Is there a reason that the test suite compile will fails?

**Comment by MaartenBent** [comment]
The most obvious one is that you use `test/test.rc` instead of `tests/test.rc`.
You could have at least posted the relevant error message, maybe you would have noticed it yourself too.

**Comment by oneeyeman1** [comment]
@MaartenBent ,
Thank you Typo fixed.

@vadz ,
Do you want me to reference the icon from the samples directory or copy it to the tests?

Thank you

**Comment by oneeyeman1** [comment]
This is superceeded by 23558
